docs/tutorials/mastering.md

```diff
@@ -899,50 +904,191 @@ Due to the newgrp command you will need to log out twice as it started a new she
+}
+
+EOF
+```
+
+Initialize Juju Terraform provider:
+
+```console
+terraform init
 ```
 
-Deploy the control plane Juju bundle:
+Deploy SD-Core Control Plane:
 
 ```console
-juju deploy sdcore-control-plane-k8s --trust --channel=beta --overlay control-plane-overlay.yaml
+terraform apply -auto-approve
 ```
 
-Expose the Software as a Service offer for the AMF.
+Monitor the status of the deployment:
 
 ```console
-juju offer control-plane.amf:fiveg-n2
+watch -n 1 -c juju status --color --relations
 ```
 
-### Checkpoint 3: Does AMF External load balancer service exist ?
+The deployment is ready when all the charms are in the `Active/Idle` state. 
+It is normal for `grafana-agent` to remain in waiting state.
 
-You should be able to see the AMF External load balancer service in Kubernetes. Log in to the `control-plane` VM and execute the following command:
+Once the deployment is ready, we will proceed to the configuration part.
+
+Log out of the VM.
+
+Get the IP addresses of the AMF and Traefik LoadBalancer services:
+
+Log in to the `control-plane` VM:
 
 ```console
 multipass shell control-plane
+```
+
+Get LoadBalancer services:
+
+```console
+microk8s.kubectl get services -A | grep LoadBalancer
+```
+
+This will show output similar to the following:
+
+```console
+control-plane    amf-external  LoadBalancer  10.152.183.179  10.201.0.52   38412:30408/SCTP
+control-plane    traefik       LoadBalancer  10.152.183.28   10.201.0.53   80:32349/TCP,443:31925/TCP
+```
+
+Note both IPs - in this case `10.201.0.52` for the AMF and `10.201.0.53` for Traefik. 
+We will need them shortly. 
+
+Log out of the VM.
+
+Log into the `juju-controller` VM:
+
+```console
+multipass shell juju-controller
+```
+
+Configure AMF external IP, using the address obtained in the previous step. 
+To do that, edit `amf_config` in the `main.tf` file in the `terraform` directory:
+
+```console
+cd terraform
+```
+
+Updated `amf_config` should look like similar to the below:
+
+```
+(...)
+module "sdcore-control-plane" {
+  (...)
+  amf_config = {
+    external-amf-ip       = "10.201.0.52"
+    external-amf-hostname = "amf.mgmt"
+  }
+  (...)
+}
+(...)
+```
+
+Configure Traefik's external hostname, using the address obtained in the previous step.
+To do that, edit `traefik_config` in the `main.tf` file.
+
+Updated `traefik_config` should look like similar to the below:
+
+```console
+(...)
+module "sdcore-control-plane" {
+  (...)
+  traefik_config = {
+    routing_mode      = "subdomain"
+    external_hostname = "10.201.0.53.nip.io"
+  }
+  (...)
+}
+(...)
+```
+
+Apply the changes:
+
+```console
+terraform apply -auto-approve
+```
+
+Log out of the VM.
+
+### Checkpoint 3: Does the AMF external LoadBalancer service exist?
+
+You should be able to see the AMF external LoadBalancer service in Kubernetes. 
+
+Log in to the `control-plane` VM:
+
+```console
+multipass shell control-plane
+```
+
+Get LoadBalancer services:
+
+```console
```

Comment:
We are now simply repeating what we did in L1010. Maybe this section is no longer needed?

Reply:
I left it, because it felt consistent with other steps, but I agree that it doesn't make sense.
Removed.